server/routerlicious/packages/lambdas/src/scribe/lambda.ts

```diff
@@ -445,20 +448,32 @@ export class ScribeLambda implements IPartitionLambda {
 			const checkpointReason = this.getCheckpointReason();
 			if (checkpointReason !== undefined) {
 				// checkpoint the current up-to-date state
-				this.prepareCheckpoint(message, checkpointReason);
+				if (this.isDocumentCorrupt) {
+					this.prepareCheckpoint(message, checkpointReason, true);
+				} else {
+					this.prepareCheckpoint(message, checkpointReason);
+				}
 			} else {
 				this.updateCheckpointIdleTimer();
 			}
 		}
 	}
 
-	public prepareCheckpoint(message: IQueuedMessage, checkpointReason: CheckpointReason) {
+	public prepareCheckpoint(
+		message: IQueuedMessage,
+		checkpointReason: CheckpointReason,
+		bypassKafka: boolean = false,
+	) {
 		// Get checkpoint context
 		const checkpoint = this.generateScribeCheckpoint(message.offset);
 		this.updateCheckpointMessages(message);
 
 		// write the checkpoint with the current up-to-date state
-		this.checkpoint(checkpointReason);
+		if (!bypassKafka) {
+			// kafka checkpoint
+			this.checkpoint(checkpointReason);
```

Comment:
I am not sure if these comments are correct. this.checkpoint() only updates the checkpoint info right? It does not do the checkpointing itself. That happens in checkpointCore. And checkpoint core is where both the db checkpointing and the kafka checkpointing happens

Reply:
Yes, you are correct. I added this in the wrong spot. I meant to add this on `this.context.checkpoint`. Updating now